modules/core/src/main/java/org/apache/ignite/client/IgniteClient.java

```diff
@@ -57,25 +72,50 @@
      */
     public Collection<String> cacheNames() throws ClientException;
 
+    /**
+     * @return Collection of names of currently available caches or an empty collection if no caches are available.
+     */
+    public IgniteClientFuture<Collection<String>> cacheNamesAsync() throws ClientException;
+
     /**
      * Destroy cache.
      */
     public void destroyCache(String name) throws ClientException;
 
+    /**
+     * Destroy cache.
+     */
+    public IgniteClientFuture<Void> destroyCacheAsync(String name) throws ClientException;
+
     /**
      * Create cache.
      *
      * @param name Cache name.
      */
     public <K, V> ClientCache<K, V> createCache(String name) throws ClientException;
 
+    /**
+     * Create cache.
```

Comment:
Probably, it worth to mention that the default configuration is going to be used.

Reply:
Fixed